src/top-navigation/parts/utility.tsx

```diff
@@ -22,9 +23,12 @@ export interface UtilityProps {
 export default function Utility({ hideText, definition, offsetRight }: UtilityProps) {
   const hasIcon = !!definition.iconName || !!definition.iconUrl || !!definition.iconAlt || !!definition.iconSvg;
   const shouldHideText = hideText && !definition.disableTextCollapse && hasIcon;
-  const ariaLabel = definition.ariaLabel ?? definition.text;
+  let ariaLabel = definition.ariaLabel ?? definition.text;
 
   if (definition.type === 'button') {
+    ariaLabel = definition.ariaLabel
+      ? definition.ariaLabel
+      : joinStrings(definition.text, definition.externalIconAriaLabel);
```

Comment:
I think we should avoid "building" any strings for alternative text because we don't know the implications in some localizations.

We have the `externalIconAriaLabel` property on `Link`/`InternalLink`, right? I think it would make more sense to pass that down. What do you think?

Reply:
I see. The whole responsiveness logic makes this a little unintuitive either way we go. I'm fine with this approach then.